### DISCUSSION TITLE
How to conduct multi-column SEARCH using different index that built on corresponding vector column.

### DISCUSSION THREAD

**Discussion by Lorisyy** [milvus-io/milvus#12085] [Q&A and General discussion] [answered]
If the ordinary search is conducted like this:
```Python3
vec = [0.1,0.2]  

results = collection.search(data=[vec], anns_field="vec_col_1", param=search_params, limit=20, expr=None)  
```
Which is equivalent to this SQL:
```PostgreSQL
SELECT id, vec_col_1<>[0.1,0.2]  

FROM table_name  

ORDER BY vec_col_1<>[0.1,0.2]  

LIMIT 20  
```
So, how to conduct a multi-vector search like this SQL:
```PostgreSQL  
SELECT id, (vec_col_1<>[0.1,0.2])+(vec_col_2<>[0.3，0.4])  

FROM table_name  

ORDER BY (vec_col_1<>[0.1,0.2])+(vec_col_2<>[0.3，0.4])  

LIMIT 20  
```

**Comment by Lorisyy** [comment]
Can just anybody please answer my aforementioned question?

**Comment by yhmo** [comment]
ANN search is not as same as SQL search. 
For ANN search, the result is ordered by similarity, so we don't need 'ORDER BY'. We can say the "topK" is similar to "limit". 
For example: 
`collection.search(data=[vec], anns_field="vec_col_1", param=search_params, topK=20, expr="col_2 > 100", output_fields=["col_2, col_3"])`
 is like 
"do ANN search in vec_col_1 by [vec]" and "SELECT col_2, col_3 FROM collection WHERE col_2 > 100 LIMIT 20"

**Reply by Lorisyy** [reply]
OKay, thanks for your answer :)

**Comment by yhmo** [comment, marked as answer]
Currently milvus don't support multi-vector-fields search.
We need some kind of metric machinery to support multi-vector-fields search. This is a long term feature, not yet plan to do this now.

**Reply by Lorisyy** [reply]
I see, thanks a lot for your help! 
Looking forward to your update on this, nice work!
Have a good day.

**Reply by Lorisyy** [reply]
I'll try to implement this somehow... If you are interested, plz contact me via Github @Lorisyy and we can discuss more.

**Comment by Lorisyy** [comment]
I'll try to implement this somehow... If anyone is interested, plz contact me via Github @Lorisyy and we can discuss more.
Best regards.

**Reply by twvacek** [reply]
Possibly related...A feature I would be interested is allowing wildcard values within a vector query. Based on my limited knowledge, it seems that quantization indices could support this (in L2 metric) because the diameter of the cluster of vectors represented does not increase, although the index would be less effective since more clusters have to be broken open. I can also think of greedy ways to upper bound the reduced cluster diameter.

**Reply by xiaofan-luan** [reply]
> I'll try to implement this somehow... If anyone is interested, plz contact me via Github @Lorisyy and we can discuss more. Best regards.

Any progress on this? we are intrested to bring in multi vector search into milvus

**Reply by xiaofan-luan** [reply]
> Possibly related...A feature I would be interested is allowing wildcard values within a vector query. Based on my limited knowledge, it seems that quantization indices could support this (in L2 metric) because the diameter of the cluster of vectors represented does not increase, although the index would be less effective since more clusters have to be broken open. I can also think of greedy ways to upper bound the reduced cluster diameter.

when we talkabout wildcard values, do we mean search vectors like [1.0, 2.0, *, *, 3.0]?what is the use case? 
We can support user defined distance metrics to support the metrics, but I'm wondering the use case

**Reply by Lorisyy** [reply]
> twvacek

Hi! Could you plz specify your use case to clarify your motivation and goal (like I did in this topic)? Thanks!

**Reply by Lorisyy** [reply]
> > I'll try to implement this somehow... If anyone is interested, plz contact me via Github @Lorisyy and we can discuss more. Best regards.
> 
> Any progress on this? we are intrested to bring in multi vector search into milvus

Yes, I do have some ideas but without the full implementations.

**Reply by xiaofan-luan** [reply]
> 
They way we are thinking of is to implement user specific distance across two vector columns.
then user could search across multiple columns with the specific index

**Reply by twvacek** [reply]
The wildcard idea is a special case of multivector idea, but with the goal of having a single efficient index. (I have no knowledge of what will be performant or not, just thoughts.) The use case I had in mind is a record linkage problem with missing fields. The vector space can be a direct sum of subspaces (each vector would be a concatenation of sub-vectors). Because of missing data, we don't know where to look in some of the subspaces. The number of combinations of present and missing fields is intractable. This problem setting could also be solved by multi-vector search.

**Reply by dee-corr** [reply]
Hello, just wondering if there is any update on this? I'm also looking for this feature
